### DISCUSSION TITLE
ESP32 Version 2.03

### DISCUSSION THREAD

**Discussion by tomryan2** [Dlloydev/ESP32-ESP32S2-AnalogWrite#17] [Q&A]
Will this library run on esp32 version 2.03? The example analogWriteTest sketch fails to initialize ledC and the Fade sketch doesn't compile. (version 2.03 contains its own analogWrite() fx which causes ambiguity in the overloaded fxs called in Fade)

**Comment by tomryan2** [comment]
Aha I just had to change to version 2.0.0. Thank you for the wonderful library!

*Edit: I have noticed that phase control does not work properly with analogWrite(pin, duty, 17000, 8, 0). The duty cycles appear a bit low and the phase is offset a great deal. I suppose this is unsurprising given that 17 KHz is well above the tested range.

**Reply by Dlloydev** [reply]
Thank you for your feedback! 
I'll take a look at making this compatible with the newer ESP32 versions after seeing there's still some interest.
Quite busy right now, but should have an updated version ready within a few weeks.

**Comment by Dlloydev** [comment]
Major revision 3.0.0 released.
- Compatible with latest ESP32 Arduino core
- Code refactored, now in a Class
- This is now called `pwmWrite`
- Wokwi [example](https://wokwi.com/projects/334722465700774482) added
